### ISSUE TITLE
Solve Identity Issues

### ISSUE THREAD

**Issue by OneManEquipe** [OneManEquipe/aitools#20]
Currently a `LogicObject` is something that has an automatically generated ID.

Very nice, right? WRONG.

The reason I gave IDs to every `LogicObject` was because I was thinking "this would allow more efficient operations, since most of the time I could just base everything on the ID".

But this is plainly false :P I always compare by value, and IDs are just a hindrance, as it becomes non-trivial to decide whether an object is already persisted in a storage or not (yeah, I'm working on indexing and persistence right now).

So, **step 1: remove IDs**, except for `Constant`s and `Variable`s, since their ID is actually their value.

And this brings me to the second issue, which has always been in the back(log) of my mind: languages.

I want to define the concept of a "language", which is nothing more than an ID source. `Constant`s and `Variable`s  (from now on "symbols") can then have a <Language, ID> pair as their identifiers, where Language is the identifier of the language (I'm picturing a uuid in my mind), and ID is a sequence number in that language.

This allows for the following cool things:

1. if every time I run the system I create a new language (i.e. a new uuid), no collisions between newly generated symbols will ever be possible
2. the same holds for multi-process, if I use a different language for each process
3. this does not prevent from having a common language for common symbols, like logic operators, since the uuid for that language could simply be built-in

So **step 4: restructure IDs for symbols** (yup, that's step 4)

Finally, since I'm calling them "symbols", I should create a common class for them, and put the ID functionality there.

This in turn requires an enumeration for symbol types.

Will this be better? Who knows! :D 

Anyway **step 3: give symbols a real identity with the `Symbol` class** (wait, where's step 2?)

So, to summarize:

- [x] remove ids from LogicObjects
- [x] clean up the custom pickle behaviour I'll have put in place to exclude IDs from serialization (ahhh there's step 2!)
- [x] create the `Symbol` class (with languages & co)
- [x] restructure IDs for `Symbol`s
